File: b_rank_skillcheck_archive/kimariji_05.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;
using static System.Math;

internal class Program
{
    static void Main()
    {
        int o = 0;
        int[] n = ReadLine().Split().Select( int.Parse ).ToArray();
        string[] s = new string[n[0]];

        n[1]--;
        foreach( int i in Range( 0, n[0] ) ) {
            s[i] = ReadLine();
        }
        if( n[0] == 1 ) {
            WriteLine( s[0][0] );
            return;
        }
        foreach( int i in Range( 0, n[0] ) ) {
            if( i == n[1] ) {
                continue;
            }

            int t = 0;

            foreach( int j in Range( 0, Min( s[n[1]].Length, s[i].Length ) ) ) {
                t++;
                if( s[i][j] != s[n[1]][j] ) {
                    break;
                }
            }
            o = Max( o, t );
        }
        WriteLine( s[n[1]].Substring( 0, o ) );
        return;
    }
}

File: b_rank_skillcheck_archive/kimariji_05.py
# 実行時間 10ms
def main():
    o = 0
    n, m = list(map(int, input().split()))
    s = []

    m -= 1
    for _ in [0] * n:
        s.append(input())
    if n == 1:
        return(s[0][0])
    for i in range(n):
        if i == m:
            continue

        t = 0

        for j in range(min(len(s[m]), len(s[i]))):
            t += 1
            if s[i][j] != s[m][j]:
                break
        if o < t:
            o = t
    return(s[m][:o])

print(main())
